resource-managers/kubernetes/docker/src/main/dockerfiles/spark/entrypoint.sh

```diff
@@ -109,9 +81,16 @@ case "$SPARK_K8S_CMD" in
     )
     ;;
 
+  init)
+    CMD=(
+      "$SPARK_HOME/bin/spark-class"
+      "org.apache.spark.deploy.k8s.SparkPodInitContainer"
+      "$@"
```

Comment:
any place in this entrypoint where $@ is being referenced by a spark-on-k8s command still needs the shift that was taken out on line 44, correct?  For the unknown command case we just want a passthrough, but the main case statement historically is expecting the spark-on-k8s command to be stripped. The easiest/clearest way to preserve this might just be an extra case statement like this at line 44:

```bash
case "$SPARK_K8S_CMD" in
    driver | executor | init)
       shift 1
       ;;
esac